### DISCUSSION TITLE
Yarn command to check for unused dependencies

### DISCUSSION THREAD

**Discussion by benatshippabo** [yarnpkg/berry#2718] [Ideas]
It would be nice if Yarn 2+ had some method to find unused dependencies listed in `package.json`s. Similar to `depcheck`, but also analyzing the executables run by npm scripts and peer dependencies. 

I feel like this may easily fit within `@yarnpkg/doctor`, since the package already figures out the list of all required dependencies and performs some form of analysis on the `package.json`

**Comment by wdfinch** [comment]
you should make a feature request.

**Reply by benatshippabo** [reply]
Done! https://github.com/yarnpkg/berry/issues/2744#issue-856157362
